File: src-tauri/src/commands/twitch/clips.rs
use crate::dto::twitch::clip::CreatedClip;
use crate::services;
use crate::services::twitch::get_token;
use serde::Deserialize;

#[derive(Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreateClipParams {
    pub broadcaster_id: String,
    pub title: Option<String>,
    pub duration: Option<f64>,
}

#[tauri::command]
pub async fn create_clip(
    app: tauri::AppHandle,
    params: CreateClipParams,
) -> Result<CreatedClip, String> {
    let token = get_token(&app).await?;
    services::twitch::clips::create_clip(
        &token,
        params.broadcaster_id,
        params.title,
        params.duration,
    )
    .await
}

File: src/lib/api/twitch/clips.ts
import { invokeCommand, type InvokeOptions } from "../utils.ts";

export type CreateClipParams = {
  broadcasterId: string;
  title?: string;
  duration?: number;
};

export type CreatedClip = {
  id: string;
};

export function createClip(
  params: CreateClipParams,
  options?: InvokeOptions,
): Promise<CreatedClip> {
  return invokeCommand("create_clip", params, options);
}
